data/dataset.py

```diff
@@ -187,10 +185,11 @@ def from_path(cls, data_root):
         # and accumulates image configs as it traverses dataset
         image_configs = {}
         def process_dir(files, parent_globals):
-            global_cfg = parent_globals.merge(Dataset.__global_cfg(files))
-            local_cfg = Dataset.__local_cfg(files)
+            fileset = {os.path.basename(f): f for f in files}
```

Comment:
Build a dictionary of basename to file path so config loading functions don't have to scan the list repeatedly.